tf_files/aws/modules/s3-bucket/cloud.tf

```diff
@@ -151,3 +151,94 @@ resource "aws_iam_instance_profile" "mybucket_writer" {
   name = "bucket_writer_${local.clean_bucket_name}"
   role = "${aws_iam_role.mybucket_writer.id}"
 }
+
+
+
+############################## Added by Fauzi fauzi@uchicago.edu
+
+# let's send the logs to cloudwatch as well
+
+data "aws_cloudwatch_log_group" "logs_destination" {
+  name = "${var.environment}"
+}
+
+# create the policy and role for the role that will be attached to 
+# the trail so it can write cloudwatch
+
+resource "aws_iam_role" "cloudtrail_writer" {
+  name = "cwl_writer_${local.clean_bucket_name}"
+  path = "/"
+
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "sts:AssumeRole",
+            "Principal": {
+               "Service": "cloudtrail.amazonaws.com"
+            },
+            "Effect": "Allow",
+            "Sid": ""
+        }
+    ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "trail_policy" {
+  statement {
+    effect    = "Allow"
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = ["${data.aws_cloudwatch_log_group.logs_destination.arn}"]
+  }
+
+}
+
+resource "aws_iam_policy" "trail_writer" {
+  name        = "trail_write_to_cw_${data.aws_cloudwatch_log_group.logs_destination.name}"
+  description = "Read or write     ${data.aws_cloudwatch_log_group.logs_destination.name}"
+  policy      = "${data.aws_iam_policy_document.trail_policy.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "trail_writer_role" {
+  role       = "${aws_iam_role.cloudtrail_writer.name}"
+  policy_arn = "${aws_iam_policy.trail_writer.arn}"
+}
+
+
+# first we need to create a trail in cloudtrail
+
+resource "aws_cloudtrail" "logger_trail" {
+  name                          = "${var.environment}-trail"
```

Comment:
What happens if we have multiple buckets - will they all create the same trail?

Reply:
I wasn't counting on having multiple buckets for the same commons, but it makes sense to have the module ready for multiple buckets.

Reply:
pushed a change that should take care of this in the future.